EventsExpress.Test/BuilderTests/FilterBuilderTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Linq;
+using System.Reflection;
+using EventsExpress.Core.Builders;
+using NUnit.Framework;
+
+namespace EventsExpress.Test.BuilderTests
+{
+    [TestFixture]
+    public class FilterBuilderTests
+    {
+        private FilterBuilder<int> _builder;
+        private FieldInfo _conditionField;
+        private FieldInfo _queryableField;
+
+        private bool FilterWillApply
+        {
+            get => (bool)_conditionField.GetValue(_builder);
+            set => _conditionField.SetValue(_builder, value);
+        }
+
+        private IQueryable<int> Queryable
+        {
+            get => (IQueryable<int>)_queryableField.GetValue(_builder);
+        }
+
+        [SetUp]
+        protected void Initialize()
+        {
+            var queryable = Enumerable.Range(1, 10).AsQueryable();
+            _builder = new FilterBuilder<int>(queryable);
+
+            var builderType = _builder.GetType();
+            const BindingFlags flags = BindingFlags.NonPublic | BindingFlags.Instance;
+
+            _conditionField = builderType.GetField("_filterWillApply", flags);
```

Comment:
Generally, unit tests test the interfaces, not internal implementation details. This allows to avoid unit test changes if internals change for some reason. Even though access to these internals can provide more control, I suggest we avoid doing that and create tests using only interfaces defined.

Reply:
Agreed, I will modify the tests to avoid accessing private fields